### PR TITLE
feat: Integrate Github Actions Source Client to CLI

### LINKED ISSUE #676: Github Actions Client: integrate Source Client to CLI
**Author the user story for this feature**
As a user, I want to use ci_integrations CLI to import data from Github projects so I can view metrics using the metrics web app.

**Is your feature request related to a problem? Please describe.**
Integrate data from Github projects.

**Describe the solution you'd like**
Integrate Github Actions Client to ci_integrations CLI.
Make sure that it can be configured with YAML.
All proper Factories/Parties/Sources are created.

**Additional context**
[CI integrations module architecture](https://github.com/platform-platform/monorepo/blob/master/metrics/ci_integrations/docs/01_ci_integration_module_architecture.md)

### PR DESCRIPTION
**Pull request by solid-daniyilyevtyushkin**
Fixes #676 

## Checklist

- [x] Documented
- [x] Unit tested

### CODE REVIEW

File: metrics/ci_integrations/lib/source/github_actions/client_factory/github_actions_source_client_factory.dart [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+class GithubActionsSourceClientFactory
+    implements
+        SourceClientFactory<GithubActionsSourceConfig,
+            GithubActionsSourceClientAdapter> {
+  /// Creates a new instance of the [GithubActionsSourceClientFactory].
+  const GithubActionsSourceClientFactory();
+
+  @override
+  GithubActionsSourceClientAdapter create(
+    GithubActionsSourceConfig config,
```
Comment (solid-vladlaba) [inline metrics/ci_integrations/lib/source/github_actions/client_factory/github_actions_source_client_factory.dart:22]:
```suggestion
    GithubActionsSourceConfig config
```
It seems like it fits without wrapping.

File: metrics/ci_integrations/test/source/github_actions/client_factory/github_actions_source_client_factory_test.dart [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+import 'package:test/test.dart';
+
+import '../test_utils/test_data/github_actions_config_test_data.dart';
+
+// https://github.com/platform-platform/monorepo/issues/140
+// ignore_for_file: prefer_const_constructors, avoid_redundant_argument_values
+
+void main() {
+  final githubActionsSourceConfig = GithubActionsConfigTestData.sourceConfig;
+  final githubActionsSourceClientFactory = GithubActionsSourceClientFactory();
```
Comment (solid-vladlaba) [inline metrics/ci_integrations/test/source/github_actions/client_factory/github_actions_source_client_factory_test.dart:14]:
```suggestion
  final config = GithubActionsConfigTestData.sourceConfig;
  final factory = GithubActionsSourceClientFactory();
```

File: metrics/ci_integrations/test/source/github_actions/client_factory/github_actions_source_client_factory_test.dart [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+  GithubActionsClient createClient(GithubActionsSourceConfig config) {
```
Comment (solid-vladlaba) [inline metrics/ci_integrations/test/source/github_actions/client_factory/github_actions_source_client_factory_test.dart:16]: Should we inline this function to make test cases more clear? It seems like we can move this code to the `setUp` callback.

File: metrics/ci_integrations/test/source/github_actions/client_factory/github_actions_source_client_factory_test.dart [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+
+    test(
+      ".create() creates a client with the authorization credentials from the given config",
+      () {
+        final accessToken = githubActionsSourceConfig.accessToken;
+        final authorization = BearerAuthorization(accessToken);
+
+        final client = createClient(githubActionsSourceConfig);
+
+        expect(client.authorization, authorization);
```
Comment (solid-vladlaba) [inline metrics/ci_integrations/test/source/github_actions/client_factory/github_actions_source_client_factory_test.dart:62]:
```suggestion
        expect(client.authorization, equals(authorization));
```

**Review by solid-vladlaba — CHANGES_REQUESTED** [review]
Well done, @solid-daniyilyevtyushkin! I've added a couple of suggestions below - please let me know what you think!

**Review by solid-vladlaba — APPROVED** [review]
LGTM, thanks @solid-daniyilyevtyushkin!:rocket: 
Well done! @illia-romanenko, it seems like we have the Github Actions fully integrated so we can add Metrics Web project to test integration on real data while we are working on unit tests and mock server. Coverage will be `null` for now but other moments should work.

**Comment by illia-romanenko** [comment]
Great job @solid-vladlaba @solid-daniyilyevtyushkin !

**Pull request merged**